Document/0x05d-Testing-Data-Storage.md

```diff
@@ -310,6 +406,36 @@ Inspect the source code to determine whether native Android mechanisms identify
 - Make sure that the app is using the Android KeyStore and Cipher mechanisms to securely store encrypted information on the device. Look for the patterns `AndroidKeystore`, `import java.security.KeyStore`, `import javax.crypto.Cipher`, `import java.security.SecureRandom`, and corresponding usages.
 - Use the `store(OutputStream stream, char[] password)` function to store the KeyStore to disk with a password. Make sure that the password is provided by the user, not hard-coded.
 
+##### Storing a Key - example
+
+To mitigate unauthorized use of keys on the Android device, Android KeyStore lets apps specify authorized uses of their keys when generating or importing the keys. Once made, authorizations cannot be changed.
+
+Storing a Key - from most secure to least secure:
+
+- the key is stored in hardware-backed Android KeyStore
+- all keys are stored on server and are available after strong authentication
+- master key is stored on server and use to encrypt other keys, which are stored in Android SharedPreferences
+- the key is derived each time from a strong user provided passphrase with sufficient length and salt
+- the key is stored in software implementation of Android KeyStore
+- master key is stored in Android Keystore and use to encrypt other keys, which are stored in SharedPreferences
+- all keys are stored in SharedPreferences
```

Comment:
this should be "not recommended"

Reply:
all these 4 points (from key derived from passphrase to all keys in SharedPreferences) or only 'all keys in SharedPreferences'?

Reply:
"all keys are stored in SharedPreferences" shouldn't be recommended. 
As long as keys are in the KeyStore then it's best practice, as we are suggesting to leverage a strong encryption mechanisms offered by  Android.

Reply:
Done. :)